pkg/status/mutate_condition_status.go

```diff
@@ -8,15 +8,31 @@ import (
 	kuberecorder "k8s.io/client-go/tools/record"
 )
 
-// MarkNotReady sets the condition status of an Object to Not Ready.
+// MarkNotReady sets the condition status of an Object to `Not Ready`.
 func MarkNotReady(recorder kuberecorder.EventRecorder, obj conditions.Setter, reason, msg string) {
+	conditions.Delete(obj, meta.ReconcilingCondition)
 	conditions.MarkFalse(obj, meta.ReadyCondition, reason, msg)
 	event.New(recorder, obj, eventv1.EventSeverityError, msg, nil)
 }
 
-// MarkAsStalled sets the condition status of an Object to Stalled.
+// MarkUnknown sets the condition status of an Object to `Unknown`.
+func MarkUnknown(recorder kuberecorder.EventRecorder, obj conditions.Setter, reason, msg string) {
```

Comment:
I would have it like 

```go
MarkUnknown(recorder kuberecorder.EventRecorder, metav1 condition, obj conditions.Setter, reason, msg string) {
```

Also I dislike unkown. I think we can infer a reason based on the reconciliation step at least.

Reply:
Fair enough. I'll remove it. :)